quinn-proto/src/crypto/rustls.rs

```diff
@@ -364,42 +431,24 @@ impl crypto::PacketKey for PacketKey {
         header: &[u8],
         payload: &mut BytesMut,
     ) -> Result<(), CryptoError> {
-        if payload.len() < self.key.algorithm().tag_len() {
-            return Err(CryptoError);
-        }
-
-        let payload_len = payload.len();
-        let aad = aead::Aad::from(header);
-        let nonce = self.iv.nonce_for(packet);
-        self.key.open_in_place(nonce, aad, payload.as_mut())?;
-        payload.truncate(payload_len - self.key.algorithm().tag_len());
+        let plain = self
+            .decrypt_in_place(packet, &*header, payload.as_mut())
+            .map_err(|_| CryptoError)?;
+        let plain_len = plain.len();
+        payload.truncate(plain_len);
         Ok(())
     }
 
     fn tag_len(&self) -> usize {
-        self.key.algorithm().tag_len()
+        self.tag_len()
```

Comment:
looks like a recursive call which will stack overflow

Reply:
It isn't; inherent impls get priority over trait impls.

Reply:
Interesting. Didn't even knew one can have the same method name as a direct impl and trait